mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/NullSavingStrategyValidation.java

```diff
@@ -0,0 +1,246 @@
+    } else {
+      return hasDaoLevelUnset(daoLevelInterface) && hasDoNotSetStatement;
+    }
+  }
+
+  private boolean hasDaoLevelDoNotSetStatement(TypeElement interfaceElement) {
+    DefaultNullSavingStrategy annotation =
+        interfaceElement.getAnnotation(DefaultNullSavingStrategy.class);
+    if (annotation != null) {
+      return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET;
+    }
+    return false;
+  }
+
+  private boolean hasDaoLevelSetToNullStatement(TypeElement interfaceElement) {
+    DefaultNullSavingStrategy annotation =
+        interfaceElement.getAnnotation(DefaultNullSavingStrategy.class);
+    if (annotation != null) {
+      return annotation.nullSavingStrategy() == NullSavingStrategy.SET_TO_NULL;
+    }
+    return false;
+  }
+
+  private boolean hasDaoLevelUnset(TypeElement interfaceElement) {
+    return interfaceElement.getAnnotation(DefaultNullSavingStrategy.class) == null;
+  }
+
+  private boolean queryHasDoNotSet(ExecutableElement v, boolean explicitSet) {
+    Query annotation = v.getAnnotation(Query.class);
+    if (annotation != null) {
+      if (explicitSet) {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET
+            && nullSavingStrategyExplicitlySet(v, Query.class);
+      } else {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET;
+      }
+    }
+    return false;
+  }
+
+  private boolean setEntityHasDoNotSet(ExecutableElement v, boolean explicitSet) {
+    SetEntity annotation = v.getAnnotation(SetEntity.class);
+    if (annotation != null) {
+      if (explicitSet) {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET
+            && nullSavingStrategyExplicitlySet(v, SetEntity.class);
+      } else {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET;
+      }
+    }
+    return false;
+  }
+
+  private boolean insertHasDoNotSet(ExecutableElement v, boolean explicitSet) {
+    Insert annotation = v.getAnnotation(Insert.class);
+    if (annotation != null) {
+      if (explicitSet) {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET
+            && nullSavingStrategyExplicitlySet(v, Insert.class);
+      } else {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET;
+      }
+    } else {
+      return false;
+    }
+  }
+
+  private boolean updateHasDoNotSet(ExecutableElement v, boolean explicitSet) {
+    Update annotation = v.getAnnotation(Update.class);
+    if (annotation != null) {
+      if (explicitSet) {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET
+            && nullSavingStrategyExplicitlySet(v, Update.class);
+      } else {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET;
+      }
+    } else {
+      return false;
+    }
+  }
+
+  private boolean queryHasSetToNullExplicitly(ExecutableElement v) {
```

Comment:
same here, these `HasSetToNullExplicitly` methods could be factored into a common method.